core/main_test.ts

```diff
@@ -1260,6 +1260,453 @@ actions:
       expect(result.compile.compiledGraph.operations[0].fileName).deep.equals("table.sql");
     });
   });
+
+  suite("Assertions as dependencies", ({ beforeEach }) => {
+    [
+      TestConfigs.bigquery,
+      TestConfigs.bigqueryWithDatasetSuffix,
+      TestConfigs.bigqueryWithNamePrefix
+    ].forEach(testConfig => {
+      let projectDir: any;
+      beforeEach("Create temporary dir and files", () => {
+        projectDir = tmpDirFixture.createNewTmpDir();
+        fs.writeFileSync(
+          path.join(projectDir, "workflow_settings.yaml"),
+          dumpYaml(dataform.WorkflowSettings.create(testConfig))
+        );
+        fs.mkdirSync(path.join(projectDir, "definitions"));
+        fs.writeFileSync(path.join(projectDir, "definitions/A.sqlx"),
+          `config {
+          type: "table",
+           assertions: {
+             rowConditions: ["test > 1"]
+          }
+        }
+        SELECT 1 as test
+      }
+      `);
+        fs.writeFileSync(path.join(projectDir, "definitions/A_assert.sqlx"),
+          `config {
+          type: "assertion",
+      }
+      select test from \${ref("A")} where test > 3`);
+      });
+
+      test("When dependOnDependencyAssertions property is set to true, assertions from A are added as dependencies", () => {
+        fs.writeFileSync(path.join(projectDir, "definitions/B.sqlx"),
+          `config {
+            type: "table",
+            dependOnDependencyAssertions: true,
+            dependencies: ["A"]
+        }
+        select 1 as btest
+        `);
+
+        const result = runMainInVm(coreExecutionRequestFromPath(projectDir));
+
+        expect(result.compile.compiledGraph.graphErrors.compilationErrors).deep.equals([]);
+        expect(asPlainObject(result.compile.compiledGraph.tables[1].dependencyTargets.length)).equals(3)
```

Comment:
Nit: to keep these tests flexible, we should probably do something like `result.compile.compiledGraph.tables.find(table => table.name === "A")` instead